Fix #131: ensure absolute workspace root URI

cdamus opened the pull request:
***Note** that this change builds on PR #128 for other fixes in URI computations, which should be merged before this.*

edgarmueller committed:
Remove decode overload which expected RS and Model URI

Refactor ModelRepository#addModel to not only get resource, but
actually also create it

cdamus committed:
Fix #69 Adapt URI

- make adaptUri() method more robust (e.g. for spaces in path)
- also fix #126 command deserialization with absolute resource URI

Signed-off-by: Christian W. Damus <give.a.damus@gmail.com>

cdamus committed:
Fix #131: ensure absolute workspace root URI

Signed-off-by: Christian W. Damus <give.a.damus@gmail.com>

cdamus committed:
Address code review comments #128

- common implementation of decode overloads
- fix parameter ordering in the overload
- throw exception instead of hiding exception in
  the resource that the client will never see (which
  then also aligns with the XmiCodec)
- pull in fix for local resource creation in JUnit test
   from PR #132

Signed-off-by: Christian W. Damus <give.a.damus@gmail.com>

edgarmueller commented on com.eclipsesource.modelserver.emf/src/test/java/com/eclipsesource/modelserver/emf/configuration/ServerConfigurationTest.java:
```diff
@@ -0,0 +1,69 @@
+package com.eclipsesource.modelserver.emf.configuration;
```
Header missing

edgarmueller commented on com.eclipsesource.modelserver.emf/src/test/java/com/eclipsesource/modelserver/emf/configuration/ServerConfigurationTest.java:
```diff
@@ -0,0 +1,69 @@
+		serverConfiguration.setWorkspaceRootURI(URI.createFileURI(cwd.getAbsolutePath()));
+		assertThat(serverConfiguration.getWorkspaceRoot(), is(cwd.getAbsolutePath()));
+		
+		serverConfiguration.setWorkspaceRoot(".");
```
Could we move this into its own test?

edgarmueller reviewed (commented):
LGTM! Two minor remarks

cdamus committed:
Address comments on PR #132

- add missing copyright header
- split tests for setting workspace root by path and by URI

Signed-off-by: Christian W. Damus <give.a.damus@gmail.com>

ndoschek reviewed (approved):
:+1:

ndoschek committed:
Merge branch 'master' into fix/113-workspace-root
